src/odemis/acq/test/fastem_test.py

```diff
@@ -17,16 +18,38 @@
 
 You should have received a copy of the GNU General Public License along with
 Odemis. If not, see http://www.gnu.org/licenses/.
-'''
+
+"""
+from __future__ import division
+
 import logging
-from odemis import model
-import odemis
-from odemis.acq import stream, fastem
-from odemis.util import test, img
+import math
 import os
 import time
 import unittest
+from concurrent.futures._base import CancelledError
+from datetime import datetime
+
+import numpy
+
+import odemis
+from odemis import model
+from odemis.acq import fastem, stream
+from odemis.util import test, img
 
+# Accept three values for TEST_NOHW
+# * TEST_NOHW = 1: not connected to anything => skip most of the tests
+# * TEST_NOHW = sim: technolution_asm_simulator/simulator2/run_the_simulator.py
+# * TEST_NOHW = 0 (or anything else): connected to the real hardware
+TEST_NOHW = os.environ.get("TEST_NOHW", "0")  # Default to Hw testing
+if TEST_NOHW == "sim":
+    pass
```

Comment:
You don't seem to use "sim" anywhere (I assume because the test cases should behave the same whether it's running on a simulator or on the real hardware). So you could simplify to the usual True/False.